packages/skeleton-svelte/src/lib/components/Tab/Tabs.svelte

```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+	import { setContext } from 'svelte';
+	import type { TabsProps } from './types.js';
+
+	let {
+		// A11Y
+		listLabelledBy = '',
+		panelLabelledBy = '',
+		// Root
+		base = '',
+		spaceY = 'space-y-4',
+		classes = '',
+		// Tab list
+		listBase = 'flex overflow-x-auto hide-scrollbar',
+		listSpaceX = 'space-x-4',
+		listJustify = 'justify-start',
+		listBorder = 'border-b border-surface-50-950',
+		listClasses = '',
+		// Tab panel
+		panelBase = '',
+		panelClasses = '',
+		// Tab Controls
+		controlsBase = 'flex',
+		controlsText = 'hover:text-primary-600-400',
+		controlsJustify = 'justify-center',
+		controlsActive = 'border-b-2 border-surface-950-50 text-surface-950-50',
+		controlsInactive = 'text-surface-600-400 fill-surface-600-400',
+		controlsBackground = '',
+		controlsPadding = 'px-4 py-2',
+		controlsRounded = '',
+		controlsSpacingY = 'space-y-1',
+		controlsCursor = 'cursor-pointer',
+		controlsClasses = '',
+		// Events
+		onclick = () => {},
+		onkeypress = () => {},
+		onkeydown = () => {},
+		onkeyup = () => {},
+		onchange = () => {},
+		// Snippets
+		list,
+		panel
+	}: TabsProps = $props();
+
+	setContext('base', controlsBase);
+	setContext('text', controlsText);
+	setContext('justify', controlsJustify);
+	setContext('active', controlsActive);
+	setContext('inactive', controlsInactive);
+	setContext('background', controlsBackground);
+	setContext('padding', controlsPadding);
+	setContext('rounded', controlsRounded);
+	setContext('spacingY', controlsSpacingY);
+	setContext('cursor', controlsCursor);
+	setContext('classes', controlsClasses);
+	setContext('onclick', onclick);
+	setContext('onkeypress', onkeypress);
+	setContext('onkeydown', onkeydown);
+	setContext('onkeyup', onkeyup);
+	setContext('onchange', onchange);
```

Comment:
I see that we're utilizing context a bunch, but all these properties won't be reactively set through context.
If the user would change the `controlsBase` for example that change wouldn't get to all the consumers of this context.

[Here is an example of reactive vs non-reactive context in svelte 5](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAACpWSTWvDMAyG_4pnBkkha-5pFhhlx42x67xDlqidIZGNrZSO4P8-YidNuo9CT0avXknWg3q-kw1Ynr31HMsWeMYftOYJpy89BPYADQFPuFWdqQYlt5WRmgqBgmSrlSH2UhpAYjujWhat0xCuQ2m0WRi3n7KpTz4fLW0oqAFih7LpgN2zW0slQRw91pLYE9xEq43APJ3nYy5Rd8Q-JNZZqEqDPv6o96Lzf839uCIPY4uhU3AVPOGtquVOQs0zMh245AQjWK7k0TMLtFVIcCTmxn2XOKZN-3FXN2yrjdI2Xo0g5g5xhArvDJQVyQNESaj5y7fw9H4M7Sec8WqSBBmgzmDQg-aGx_3CKzC3jSKWXkTkeV5NaH-REAqqFNr5Fvb_4fAcRnNFxx_WM9v56ejiedEnm24lT3Xhs69zpqLjes7-ZvHuvgEU4SWJRwMAAA==)

We should probably all get together and defined some guidelines around this.

Reply:
@Mahmoud-zino I'm partially with Hugo here, we have a lot of context values. The issue is we're trying to feed everything down through the parent. Where as we should localize those props to come in on the child components themselves.

Yes, there's some benefits to having a single entry point from from a consuming developer's perspective, but this at the cost of our code maintainability. This was a pattern we followed in v2 that I don't think we should continue forward in v3.

My suggestion as a rule of thumb is: if the parent doesn't use the value, it should not contain a prop for it.

As per reactivity. We shouldn't add this just because we can. We should add this only when it's absolutely required given the extra hoops required to make it work. To me, most style prop don't need to be reactive. That's a luxury, not a necessity. Honestly same goes for even props. I can't think of a lot of scenarios where your actual handler function will change, as opposed to just handling the result within the handler in a different manner.